src/mountscript.sh

```diff
@@ -741,243 +167,60 @@ parse_arguments()
-    fi
-    
-    pecho "If the problem persists, contact Microsoft support."
+if [ $nfs_vers == "4.1" ]; then
+    AZ_PREFIX="file"
+elif [ $nfs_vers == "3" ]; then
+    AZ_PREFIX="blob"
+else
+    eecho "NFS version is not supported by mount helper: $nfs_vers!"
     exit 1
 fi
 
-nfs_host=$(get_host_from_share "$1")
+nfs_host=$(get_host_from_share "$1" "$AZ_PREFIX")
 if [ $? -ne 0 ]; then
     echo "$nfs_host"
     exit 1
 fi
 
-if ! is_valid_blob_fqdn "$nfs_host"; then
-    eecho "Not a valid Azure Blob NFS endpoint: ${nfs_host}!"
-    eecho "Must be of the form 'account.blob.core.windows.net'!"
+# TODO: Comment out below code for devfabric. 'is_valid_fqdn' will fail on devfabric.
+if ! is_valid_fqdn "$nfs_host" "$AZ_PREFIX"; then
+    eecho "Not a valid Azure $AZ_PREFIX NFS endpoint: ${nfs_host}!"
+    eecho "Must be of the form 'account.$AZ_PREFIX.core.windows.net'!"
+    eecho "For isolated environments, must set the environment variable AzureEndpointOverride to the appropriate endpoint!"
     exit 1
 fi
 
-# Resolve the IP address for the NFS host
-nfs_ip=$(resolve_ipv4_with_preference_to_mountmap "$nfs_host")
-status=$?
-if [ $status -ne 0 ]; then
-    if [ $status -eq 2 ]; then
-        vecho "Resolved IP address for FQDN from mountmap [$nfs_host -> $nfs_ip]"
-    else
-        echo "$nfs_ip"
-        eecho "Cannot resolve IP address for ${nfs_host}!"
-        eecho "Mount failed!"
-        exit 1
-    fi
-fi
-
-nfs_dir=$(get_dir_from_share "$1")
+nfs_dir=$(get_dir_from_share "$1" "$AZ_PREFIX")
 if [ $? -ne 0 ]; then
     echo "$nfs_dir"
     exit 1
 fi
 
 if [ -z "$nfs_dir" ]; then
     eecho "Bad share name: ${1}!"
-    eecho "Share to be mounted must be of the form 'account.blob.core.windows.net:/account/container'!"
+    eecho "Share to be mounted must be of the form 'account.$AZ_PREFIX.core.windows.net:/account/container'!"
     exit 1
 fi
 
-mount_point="$2"
-
-OPTIONS=
-MOUNT_OPTIONS=
-
-parse_arguments $*
-
-#
-# Fix MOUNT_OPTIONS if needed.
-#
-if [ "$AZNFS_FIX_MOUNT_OPTIONS" == "1" ]; then
-    fix_mount_options
-fi
-
-#
-# Check azure nconnect flag.
-#
-if [ "$AZNFS_CHECK_AZURE_NCONNECT" == "1" ]; then
-    if ! check_nconnect; then
-        eecho "Mount failed!"
-        exit 1
-    fi
-fi
-
-#
-# Fix dirty bytes config if needed.
-#
-if [ "$AZNFS_FIX_DIRTY_BYTES_CONFIG" == "1" ]; then
-    fix_dirty_bytes_config
-fi
-
-#
-# Get proxy IP to use for this nfs_ip.
-# It'll ensure an appropriate entry is added to MOUNTMAP if not already added,
-# and an appropriate iptable DNAT rule is added.
-#
-exec {fd}<$MOUNTMAP
-flock -e $fd
-#
-# With the lock held first check if adding a new mountmap entry for this account will
-# cause "accounts mounted on one client" to exceed the limit.
-#
-if check_account_count; then
-    get_local_ip_for_fqdn $nfs_host
-    ret=$?
-    account_limit_exceeded=0
+if [ "$nfs_vers" == "4.1" ]; then
+    $OPTDIR/nfsv4mountscript.sh "$MOUNT_OPTIONS" "$OPTIONS" "$nfs_host" "$nfs_dir" "$mount_point"
```

Comment:
It would be better to define two vars MOUNTSCRIPTv3 and MOUNTSCRIPTv4 in common.sh and use it here as done for other files.